src/test/java/GameFieldTest.java

```diff
@@ -0,0 +1,47 @@
+import org.junit.Test;
+
+import java.util.Collection;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class GameFieldTest {
+
+    @Test
+    public void givenGameObjectOutOfBoundary_whenAddObject_thenCorrect(){
+        GameField gameField = new GameField(new Point(10, 10));
+
+        gameField.addObject(new GameObject(new Point(9, 9), new Point(2, 2)));
+
+        Collection<GameObject> gameObjects = gameField.getGameObjects();
+
+        assertTrue(gameObjects.size() == 0);
+    }
+
+    @Test
+    public void givenIntersectingGameObject_whenAddObject_thenCorrect(){
+        GameField gameField = new GameField(new Point(10, 10));
+
+        gameField.addObject(new GameObject(new Point(4, 4), new Point(3, 5)));
+
+        gameField.addObject(new GameObject(new Point(4, 6), new Point(2, 3)));
+
+        Collection<GameObject> gameObjects = gameField.getGameObjects();
+
+        assertTrue(gameObjects.size() == 1);
```

Comment:
Please try to split test body into three parts (given, when, then) using empty lines to be able understand author's intention more clearly:

```
    @Test
    public void shouldHasOneWhenTwoIntersectingObjectsAreAdded() {
        GameField gameField = new GameField(new Point(10, 10));  // givenpart
        gameField.addObject(new GameObject(new Point(4, 4), new Point(3, 5))); // given part
        gameField.addObject(new GameObject(new Point(4, 6), new Point(2, 3))); // given part

        Collection<GameObject> gameObjects = gameField.getGameObjects(); // when part

        assertEquals(1, gameObjects.size());  // then part
    }
```

*NOTE*: but comments should not be added, i 've added them so you can understand more clearly